raiden/network/rpc/client.py

```diff
@@ -519,13 +519,22 @@ def approve(self, contract_address, allowance):
         # `self.address` is one of the participants (maybe add this logic into
         # `NettingChannel` and keep this straight forward)
 
-        transaction_hash = self.proxy.approve.transact(
+        estimated_gas = self.proxy.approve.estimate_gas(
```

Comment:
All of these could also be abstracted away in a special function. For example:

```python
def estimate_and_transact(self, callobj, *args):
    """TODO: Add a proper docstring here"""
    estimated_gas = callobj.estimate_gas(*args, startgas=self.startgas, gasprice=self.gasprice)
    estimated_gas  = min(estimated_gas * 2, GAS_LIMIT)
    transaction_hash = callobj.transact(*args, startgas=estimated_gas, gasprice=self.gasprice)
    return transaction_hash
```

And then you would call it like this:
```python
transaction_hash = self.estimate_and_transact(self.proxy.approve, contract_address, allowance)
```